internal/impl/gcp/integration_spanner_test.go

```diff
@@ -0,0 +1,313 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"cloud.google.com/go/spanner"
+	adminapi "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/redpanda-data/connect/v4/internal/impl/gcp/changestreams"
+)
+
+var (
+	spannerProjectID  = flag.String("spanner.project_id", "", "GCP project ID for Spanner tests")
+	spannerInstanceID = flag.String("spanner.instance_id", "", "Spanner instance ID for tests")
+	spannerDatabaseID = flag.String("spanner.database_id", "", "Spanner database ID for tests")
+)
+
+func spannerDatabasePath() string {
+	return fmt.Sprintf("projects/%s/instances/%s/databases/%s", *spannerProjectID, *spannerInstanceID, *spannerDatabaseID)
+}
+
+type spannerTestHelper struct {
+	client   *spanner.Client
+	tableID  string
+	streamID string
+}
+
+func newSpannerTestHelper() (spannerTestHelper, error) {
+	ctx := context.Background()
+	client, err := spanner.NewClient(ctx, spannerDatabasePath())
+	if err != nil {
+		return spannerTestHelper{}, err
+	}
+
+	ts := time.Now().UnixNano()
+	return spannerTestHelper{
+		client:   client,
+		tableID:  fmt.Sprintf("table_%d", ts),
+		streamID: fmt.Sprintf("stream_%d", ts),
+	}, nil
+}
+
+func (h spannerTestHelper) createTableAndStream() error {
+	ctx := context.Background()
+	adm, err := adminapi.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	op, err := adm.UpdateDatabaseDdl(ctx, &databasepb.UpdateDatabaseDdlRequest{
+		Database: spannerDatabasePath(),
+		Statements: []string{
+			fmt.Sprintf("CREATE TABLE %s (id INT64 NOT NULL, active BOOL NOT NULL ) PRIMARY KEY (id)", h.tableID),
+			fmt.Sprintf("CREATE CHANGE STREAM %s FOR %s", h.streamID, h.tableID),
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return op.Wait(ctx)
+}
+
+func (h spannerTestHelper) dropTableAndStream() error {
+	ctx := context.Background()
+	adm, err := adminapi.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	op, err := adm.UpdateDatabaseDdl(ctx, &databasepb.UpdateDatabaseDdlRequest{
+		Database: spannerDatabasePath(),
+		Statements: []string{
+			fmt.Sprintf("DROP CHANGE STREAM %s", h.streamID),
+			fmt.Sprintf("DROP TABLE %s", h.tableID),
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return op.Wait(ctx)
+}
+
+// Run tests with:
+// go test . -run TestSpannerInput -spanner.project_id=sandbox-rpcn -spanner.instance_id=rpcn-tests -spanner.database_id=changestreams
+func TestIntegrationSpannerInput(t *testing.T) {
+	if *spannerProjectID == "" || *spannerInstanceID == "" || *spannerDatabaseID == "" {
+		t.Skip("Skipping Spanner integration test as required flags are not set")
+	}
+
+	h, err := newSpannerTestHelper()
```

Comment:
I like the test helper, but it'd be nice to take this a step further and move more setup into a helper, it will be nice to have lots of different tests in the future here.